bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -607,12 +608,14 @@ private Optional<String> getReviewersList() {
                                    } else {
                                        var hash = review.hash();
                                        if (hash.isPresent()) {
-                                           if (!reviewCoverage.covers(review)) {
-                                               entry += " 🔄 Re-review required (review applies to [" + hash.get().abbreviate()
-                                                       + "](" + pr.filesUrl(hash.get()) + "))";
-                                           } else {
-                                               entry += " ⚠️ Review applies to [" + hash.get().abbreviate()
-                                                       + "](" + pr.filesUrl(hash.get()) + ")";
+                                           if (!hash.get().equals(pr.headHash())) {
+                                               if (!reviewCoverage.covers(review) && tooFewReviewers) {
+                                                   entry += " 🔄 Re-review required (review applies to [" + hash.get().abbreviate()
+                                                           + "](" + pr.filesUrl(hash.get()) + "))";
+                                               } else {
+                                                   entry += " ⚠️ Review applies to [" + hash.get().abbreviate()
```

Comment:
Rest assured, I will not integrate this PR until we figure out what to do with the issue you raised, Chen. That issue should be fixed either in this PR or a follow-up one.

Personally, I think if a review is stale but otherwise good, we don't need to add any emoji. It suffices to add the commit hash the review applies to, and maybe indicate how far away that commit is from HEAD, to quantify staleness: for example, "3 commits behind". Any extra commits added since that review may only be simple merges anyway.

However, if a review is stale and restricted (i.e. does not count in required reviews), we may add "⚠️". In a sense, it would provide a similar indication to that of a repo configured with:

    useStaleReviews=true
    acceptSimpleMerges=false

In this picture, the "🔄" emoji goes away completely.

Reply:
To add to my previous message.

We should keep in mind that any indication in question is primarily for a PR _author_. From my own experience as a _reviewer_, I don't normally follow a particular PR that I have reviewed to see if my review has become stale and needs an update. If the author feels I should re-review their PR, they ping me directly.

Reply:
I was thinking along similar lines. Basically there are two main cases, `useStaleReviews=(true|false)`. Here is logic that I think does roughly what we want:

```
if (useStaleReviews) {
    display the "⚠️" emoji and "Review applies to HASH" message always for stale reviews
} else {
    display the "🔄" emoji and "Re-review required" message for all stale reviews
    if (acceptSimpleMerges) {
        display the "review applies to HASH" message without any emoji for non-HEAD simple merge reviews [1]
    }
}

[1] By this I mean a review of a commits that isn't the HEAD, but that differs
from the head only by a simple merge
```

This ensures that the same PR doesn't show both the "⚠️" and "🔄" emojis.

Another possible incremental improvement we could consider is to only display the re-review needed emoji + message if the review criteria isn't satisfied. This would mean that as soon the minimum number of Reviewers re-review, the rest of the stale reviews lose the emoji (but retain a message indicating that they are out of date).

```
if (useStaleReviews) {
    display the "⚠️" emoji and "Review applies to HASH" message always for stale reviews
} else {
    if (review criteria not satisfied) {
        display the "🔄" emoji and "Re-review required" message for all stale reviews
        if (acceptSimpleMerges) {
            display the "review applies to HASH" message without an emoji for non-HEAD simple merge reviews [1]
        }
    } else {
        display the "review applies to HASH" message without an emoji for stale reviews and non-HEAD simple merge reviews [1]
    }
}
```

Reply:
> [1] By this I mean a review of a commits that isn't the HEAD, but that differs
from the head only by a simple merge

I would clarify that trailing bit as follows:

> but that differs from the head only by commits that are simple merges

This is to convey that for our purposes a series of adjacent simple merges is no different from a single simple merge.

Back to your proposal. I'll try to implement the second approach because it's in line with what this PR already does.